process-migration-service/src/main/java/org/kie/processmigration/service/impl/MigrationServiceImpl.java

```diff
@@ -0,0 +1,200 @@
+import org.kie.processmigration.service.MigrationService;
+import org.kie.processmigration.service.PlanService;
+import org.kie.processmigration.service.SchedulerService;
+import org.kie.server.api.model.admin.MigrationReportInstance;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@ApplicationScoped
+public class MigrationServiceImpl implements MigrationService {
+
+    private static final Logger logger = LoggerFactory.getLogger(MigrationServiceImpl.class);
+
+    @PersistenceContext
+    private EntityManager em;
+
+    @Inject
+    private PlanService planService;
+
+    @Inject
+    private KieService kieService;
+
+    @Inject
+    private SchedulerService schedulerService;
+
+    @Inject
+    private CredentialsService credentialsService;
+
+    @Override
+    public Migration get(Long id) {
+        TypedQuery<Migration> query = em.createNamedQuery("Migration.findById", Migration.class);
+        query.setParameter("id", id);
+        try {
+            return query.getSingleResult();
+        } catch (NoResultException e) {
+            return null;
+        }
+    }
+
+    @Override
+    public List<MigrationReport> getResults(Long id) {
+        TypedQuery<MigrationReport> query = em.createNamedQuery("MigrationReport.findByMigrationId", MigrationReport.class);
+        query.setParameter("id", id);
+        return query.getResultList();
+    }
+
+    @Override
+    public List<Migration> findAll() {
+        return em.createNamedQuery("Migration.findAll", Migration.class).getResultList();
+    }
+
+    @Override
+    @Transactional
+    public Migration submit(MigrationDefinition definition, Credentials credentials) {
+        Plan plan = planService.get(definition.getPlanId());
+        if (plan == null) {
+            throw new PlanNotFoundException(definition.getPlanId());
+        }
+        Migration migration = new Migration(definition);
+        em.persist(migration);
+        if (ExecutionType.SYNC.equals(definition.getExecution().getType())) {
+            credentialsService.save(credentials.setMigrationId(migration.getId()));
+            migrate(migration, plan, credentials);
+        } else {
+            schedulerService.scheduleMigration(migration, credentials);
+        }
+        return migration;
+    }
+
+    @Override
+    @Transactional
+    public Migration delete(Long id) {
+        Migration migration = get(id);
+        if (migration == null) {
+            return null;
+        }
+        em.remove(migration);
+        return migration;
+    }
+
+    @Override
+    @Transactional
+    public Migration update(Long id, MigrationDefinition definition, Credentials credentials) throws ReScheduleException {
+        Migration migration = get(id);
+        if (migration == null) {
+            return null;
+        }
+        if (!ExecutionStatus.SCHEDULED.equals(migration.getStatus())) {
+            throw new ReScheduleException("The migration is already started and cannot be re-scheduled");
```

Comment:
maybe change the message as I find it bit confusing

```
The migration is  not scheduled and cannot be re-scheduled
```

Reply:
Will do, thanks